### PR TITLE
Make diff smaller

### PR DESCRIPTION
**Pull request by daipom**
This is one idea of fixing [Support windows](https://github.com/daipom/fluent-plugin-watch-process/pull/2)  focusing on simplifying the difference.

I have tried to keep the current unix logic as unchanged as possible,
and to just add new Windows logic.

- [x] consider if this is better

### CODE REVIEW

**Review by fujimotos — APPROVED** [review]

File: lib/fluent/plugin/in_watch_process.rb [hunk 1]
```diff
@@ -44,53 +54,175 @@ def shutdown
...
-        !OS.windows?
+      def pipe_fixing_locale(format: "ddd MMM dd HH:mm:ss yyyy", locale: "en-US")
+        # In Windows, setting the "$env:Lang" environment variable is not effective in changing the format of the output.
+        # You can use "Datetime.ToString" method to change format of datetime values in for-each pipe.
+        # Note: About "DateTime.ToString" method: https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring
+        # Note: About "Custom date and time format strings": https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings
+        return "" unless @keys.include?("start_time")
+
+        " | %{
+          $_.StartTime = $_.StartTime.ToString(
```
Comment (fujimotos) [inline lib/fluent/plugin/in_watch_process.rb:213]: FWIW you can just do `StartTime.ToString("o")` here. You'll
get the following, regardless of what is the current locale.

```
2008-06-15T21:15:07.0000000
```

This is as per [the documentation](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-5.0#System_DateTime_ToString_System_String_)


Screenshot_2021-05-17_21-57-14

**Review by ashie — APPROVED** [review]
:+1:

File: test/plugin/test_in_watch_process.rb [hunk 2]
```diff
@@ -25,7 +25,8 @@ def test_configure
     assert_equal ['apache', 'mycron'], d.instance.lookup_user
   end
 
-  def test_emit
+  def test_unixlike
+    return if Fluent.windows?
```
Comment (kenhys) [inline test/plugin/test_in_watch_process.rb]: `omit "EXPLAIN THE REASON" if Fluent.windows?` is better if the test must be skipped.

File: test/plugin/test_in_watch_process.rb [hunk 3]
```diff
@@ -35,4 +36,75 @@ def test_emit
     d.run(expect_emits: 1, timeout: 3)
     assert(d.events.size > 1)
   end
+
```
Comment (kenhys) [inline test/plugin/test_in_watch_process.rb:39]: Grouping windows test case may be better by:

```
sub_test_case "xxxx" do 
  def test_default
  ...
  def test_customized_keys
 ...
  def test_lookup_user
 ...
end
```

File: test/plugin/test_in_watch_process.rb (hunk 2 above)
Comment (daipom) [reply test/plugin/test_in_watch_process.rb]: Thank you for the advice.
I have fixed this in https://github.com/daipom/fluent-plugin-watch-process/pull/3/commits/0fcd97df35e7d90d9dd2953bb4ad056f7003deaf.

File: test/plugin/test_in_watch_process.rb (hunk 3 above)
Comment (daipom) [reply test/plugin/test_in_watch_process.rb:39]: Thank you for the advice.
I have fixed this in https://github.com/daipom/fluent-plugin-watch-process/pull/3/commits/0fcd97df35e7d90d9dd2953bb4ad056f7003deaf.

**Pull request merged**